File: skills/invoice-conversion/examples/json-to-xlsx.js
#!/usr/bin/env node
// Convert Invapi Invoice JSON file(s) to an Excel spreadsheet
// Usage: node json-to-xlsx.js <invoice.json> [output.xlsx]
//
// Requires: INVAPI_API_KEY environment variable

const fs = require("fs");

const input = process.argv[2];
if (!input) {
  console.error("Usage: node json-to-xlsx.js <invoice.json> [output.xlsx]");
  process.exit(1);
}
const output = process.argv[3] || "invoices.xlsx";

const apiKey = process.env.INVAPI_API_KEY;
if (!apiKey) {
  console.error("Error: INVAPI_API_KEY environment variable is not set.");
  process.exit(1);
}

// Wrap the invoice in the required {"invoices": [...]} format
const invoice = JSON.parse(fs.readFileSync(input, "utf-8"));
const body = JSON.stringify({ invoices: [invoice] });

fetch("https://api.invapi.org/api/v1/json/xlsx", {
  method: "POST",
  headers: {
    "x-api-key": apiKey,
    "Content-Type": "application/json",
  },
  body,
})
  .then((res) => res.arrayBuffer())
  .then((buf) => {
    fs.writeFileSync(output, Buffer.from(buf));
    console.log(`Excel file written to ${output}`);
  })
  .catch((err) => {
    console.error(err);
    process.exit(1);
  });

File: skills/invoice-conversion/examples/json-to-xlsx.py
#!/usr/bin/env python3
# Convert Invapi Invoice JSON file(s) to an Excel spreadsheet
# Usage: python json-to-xlsx.py <invoice.json> [output.xlsx]
#
# Requires: INVAPI_API_KEY environment variable, requests library

import json
import os
import sys

import requests

if len(sys.argv) < 2:
    print("Usage: python json-to-xlsx.py <invoice.json> [output.xlsx]", file=sys.stderr)
    sys.exit(1)

input_path = sys.argv[1]
output_path = sys.argv[2] if len(sys.argv) > 2 else "invoices.xlsx"

api_key = os.environ.get("INVAPI_API_KEY")
if not api_key:
    print("Error: INVAPI_API_KEY environment variable is not set.", file=sys.stderr)
    sys.exit(1)

# Wrap the invoice in the required {"invoices": [...]} format
with open(input_path, "r") as f:
    invoice = json.load(f)

resp = requests.post(
    "https://api.invapi.org/api/v1/json/xlsx",
    headers={"x-api-key": api_key, "Content-Type": "application/json"},
    json={"invoices": [invoice]},
)
resp.raise_for_status()

with open(output_path, "wb") as f:
    f.write(resp.content)

print(f"Excel file written to {output_path}")
